src/main/java/de/tum/in/www1/artemis/repository/specs/UserSpecs.java

```diff
@@ -102,6 +102,31 @@ public static Specification<User> getInternalOrExternalSpecification(boolean int
         }
     }
 
+    /**
+     * Creates the specification to look for the user without Registration Numbers.
+     *
+     * @param noRegistrationNumber true if the account should not have a registration number
+     * @param withRegistrationNumber true if the account should have a registration number
+     * @return specification used to chain database operations
+     */
+    public static Specification<User> getWithOrWithoutRegistrationNumberSpecification(Boolean noRegistrationNumber, Boolean withRegistrationNumber) {
+        if (!noRegistrationNumber && !withRegistrationNumber) {
```

Comment:
❓Question: What happens in the case that I pass in both parameters as true?

Reply:
It is not possible due to the client side implementation (you can toggle one of the options (which are with, without and none of the filters) but not pass both as true)